doc/developer/design/20230714_show_privileges.md

```diff
@@ -0,0 +1,202 @@
+    - `SECRETS`
+    - `CONNECTIONS`
+    - `DATABASES`
+    - `SCHEMAS`
+    - `SYSTEM`
+- `FROM <schema-name>` will filter the output to only exclude objects in a schema other `<schema-name>`.
+    - If specified, top level objects, such as database, schemas, and clusters are not included.
+    - If not specified then objects are limited to the current schema and top level objects are
+    - shown.
+
+The output will include the following columns:
+
+- `name: text` The name of the object.
+- `object_type: text` The type of the object.
+- `privileges: text` Human readable version of the privileges that held by the current session.
+
+Here are some example queries:
+
+```sql
+SHOW PRIVILEGES;
+
+ name       | object_type | privileges
+------------+-------------+----------------
+ my_table   | TABLE       | SELECT, INSERT
+ my_view    | VIEW        | SELECT
+ my_cluster | CLUSTER     | CREATE
+```
+
+```sql
+SHOW PRIVILEGES ON TABLES;
+
+ name       | object_type | privileges
+------------+-------------+----------------
+ my_table   | TABLE       | SELECT, INSERT
+```
+
+```sql
+SHOW PRIVILEGES FROM public;
+
+ name       | object_type | privileges
+------------+-------------+----------------
+ my_table   | TABLE       | SELECT, INSERT
+ my_view    | VIEW        | SELECT
+```
+
+```sql
+SHOW PRIVILEGES ON VIEWS FROM public;
+
+ name       | object_type | privileges
+------------+-------------+----------------
+ my_view    | VIEW        | SELECT
+```
+
+### `SHOW DEFAULT PRIVILEGES`
+
+Will show all default privileges
+
+The syntax will be `SHOW DEFAULT PRIVILEGES [ON <object-type>]`.
+
+- `ON <object-type>` will filter the output to only include objects of type `<object-type>`. Valid
+  values are:
+  - `TABLES`
+  - `VIEWS`
+  - `MATERIALIZED VIEWS`
+  - `SOURCES`
+  - `SINKS`
+  - `TYPES`
+  - `CLUSTERS`
+  - `SECRETS`
+  - `CONNECTIONS`
+  - `DATABASES`
+  - `SCHEMAS`
+
+The output will include the following columns:
+
+- `role: text` The name of the affected role.
+- `database: text` The name of the affected database.
+- `schema: text` The name of the affected schema.
+- `object_type: text` The type of object.
+- `grantee: text` The role being granted privileges.
+- `privileges: text` Human readable version of the privileges that will be granted.
+
+Here are some example queries:
+
+```sql
+SHOW DEFAULT PRIVILEGES;
+
+ role   | database    | schema | object_type | grantee   | privileges
+--------+-------------+--------+-------------+-----------+----------------
+ PUBLIC | NULL        | NULL   | TABLE       | dev       | INSERT, SELECT
+ joe    | materialize | NULL   | VIEW        | PUBLIC    | SELECT
+ qa     | materialize | public | TYPE        | scientist | USAGE
+```
+
+```sql
+SHOW DEFAULT PRIVILEGES ON TABLES;
+
+ role   | database    | schema | object_type | grantee   | privileges
+--------+-------------+--------+-------------+-----------+----------------
+ PUBLIC | NULL        | NULL   | TABLE       | dev       | INSERT, SELECT
+```
```

Comment:
What will the output be for the object types that are not in a schema (databases, schemas, and clusters)?

Reply:
It would be the same, but `database` and `schema` would be NULL. I can add an example.

Reply:
Done.